src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -150,6 +150,36 @@ public class DataController : ControllerBase
             }
         }
 
+        /// <summary>
+        /// Create a new data object of the defined data type
+        /// </summary>
+        /// <param name="dataType">The data type id</param>
+        /// <returns>Return a new instance of the data object including prefill and initial calculations</returns>
+        [Authorize]
+        [HttpPost]
+        [DisableFormValueModelBinding]
+        [RequestSizeLimit(REQUEST_SIZE_LIMIT)]
+        [ProducesResponseType(typeof(DataElement), 200)]
+        [Route("{org}/{app}/v1/data")]
+        public async Task<ActionResult> Post([FromQuery] string dataType)
+        {
+            string classRef = _appResourcesService.GetClassRefForLogicDataType(dataType);
```

Comment:
Do we need a few validations here in case dataType is blank, does not match any type etc.? Similar to what you have on the PUT.

Reply:
added validation of dataType by checking if we can find a matching class ref for both endpoints.